### ISSUE TITLE
Adding contact/investigator as a non-admin user needs permission checking

### ISSUE THREAD

**Issue by kazoompa** [obiba/mica2#4282]
To comply to with Pentest security suggestions, adding contacts/investigators is more restrictive. 

Mica should let user add contact/investigator of accessible networks and studies.

**Comment by rwissa** [comment]
what is this?

**Comment by rwissa** [comment]
what is this and anything to do here?

**Comment by kazoompa** [comment]
No, I have added some APIs for the external editor functionality that will accommodate this need. It is not yet PR-ed or merged.
